src/sage/rings/real_mpfi.pyx

```diff
@@ -950,16 +950,16 @@ cdef class RealIntervalField_class(sage.rings.abc.RealIntervalField):
         """
         return 1
 
-    def gens(self):
+    def gens(self) -> tuple:
         """
-        Return a list of generators.
+        Return a tuple of generators.
 
         EXAMPLES::
 
             sage: RIF.gens()
-            [1]
+            (1,)
         """
-        return [self.gen()]
+        return (self.gen(),)
```

Comment:
While a `tuple` is probably a better choice here, is this not technically a breaking change? Since tuples have less capabilities than a list (since they are immutable)? Not sure what the deprecation rules would be here, or if they even apply.

Reply:
Thanks for the comments.

The aim is to make sure that all `.gens` method everywhere in sage all return either `tuple` or `Family`.

This may not be possible, but at least it is worth trying.

The annotation here is mostly to remember easily that this property has been checked. Refining this annotation would be better done in another occasion.

Reply:
I think what you wrote here addresses the other comment I made. I don't think it addresses my question of whether this is a breaking change.

Reply:
well, this is not really a breaking change. Nobody should ever try to modify the generators of some existing object. It really would be a bad idea. If somebody out there is doing that, then the change to tuple done here will help to make their code better once fixed.

Reply:
Fair enough. If someone really needs to modify the list, wrapping the call to `.gens()` with `list()` is simple enough.